File: BaseLibrary/command/handshake/InitCommand.cs
using System;
using System.Collections.Generic;
using BaseLibrary.command.common;

namespace BaseLibrary.command.handshake {
    public class InitCommand : AHandShakeCommand {
        public static int NAME_MAX_LENGTH = 10;
        public static int TEAM_NAME_MAX_LENGTH = 40;

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }


        public string NAME { get; private set; }
        public string TEAM_NAME { get; private set; }
        public RobotType ROBOT_TYPE { get; private set; }


        public InitCommand(string name) : this(name, null) {}

        public InitCommand(string name, string teamName) : this(name, teamName, RobotType.TANK) { }

        public InitCommand(string name, string teamName, RobotType robotType) {
            if (name == null) {
                throw new ArgumentNullException(nameof(name));
            }

            if (teamName == null) {
                teamName = "";
            }

            NAME = name.Substring(0, Math.Min(name.Length, NAME_MAX_LENGTH));
            TEAM_NAME = teamName.Substring(0, Math.Min(teamName.Length, TEAM_NAME_MAX_LENGTH));
            ROBOT_TYPE = robotType;
        }

    }

    public enum RobotType {
        REPAIRMAN, TANK, MINE_LAYER, NONE // none is not defined type
    }

}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/handshake/InitCommand.java

package BaseLibrary.communication.command.handshake;

import BaseLibrary.battlefield.Robot.RobotType;
import BaseLibrary.communication.protocol.AProtocol;

public class InitCommand extends AHandShakeCommand {
	
	private static final String NAME = "INIT";

	static {
		AProtocol.registerForDeserialize(NAME, InitCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	public static int NAME_MAX_length = 10;
	public static int TEAM_NAME_MAX_length = 40;

	private String name;
	private String teamName;
	private RobotType robotType;

	public InitCommand(String name) {
		this(name, null);
	}

	public InitCommand(String name, String teamName) {
		this(name, teamName, RobotType.TANK);
	}

	public InitCommand(String name, String teamName, RobotType robotType) {
		if (name == null) {
			throw new IllegalArgumentException("name cannot be null.");
		}

		if (teamName == null) {
			teamName = "";
		}

		this.name = name.substring(0, Math.min(name.length(), NAME_MAX_length));
		this.teamName = teamName.substring(0, Math.min(teamName.length(), TEAM_NAME_MAX_length));
		this.robotType = robotType;
		pending = false;
	}

	public String getName() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return name;
	}
	public String getTeamName() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return teamName;
	}
	public RobotType getRobotType() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return robotType;
	}
}
